burn-candle/src/ops/module.rs

```diff
@@ -1,24 +1,90 @@
-use burn_tensor::ops::{
-    ConvOptions, ConvTransposeOptions, MaxPool2dBackward, MaxPool2dWithIndices, ModuleOps,
+use burn_tensor::{
+    ops::{ConvOptions, ConvTransposeOptions, MaxPool2dBackward, MaxPool2dWithIndices, ModuleOps},
+    Shape,
 };
+use candle_core::ToUsize2;
 
 use crate::{
     element::{CandleElement, FloatCandleElement, IntCandleElement},
-    CandleBackend,
+    ops::base::reshape,
+    CandleBackend, CandleTensor,
 };
 
 use super::base::{FloatTensor, IntTensor};
 
 impl<F: FloatCandleElement, I: IntCandleElement> ModuleOps<CandleBackend<F, I>>
     for CandleBackend<F, I>
 {
+    fn conv1d(
+        x: FloatTensor<Self, 3>,
+        weight: FloatTensor<Self, 3>,
+        bias: Option<FloatTensor<Self, 1>>,
+        options: ConvOptions<1>,
+    ) -> FloatTensor<Self, 3> {
+        let conv = x
+            .tensor
+            .conv1d(
+                &weight.tensor,
+                options.padding[0],
+                options.stride[0],
+                options.dilation[0],
+                options.groups,
+            )
+            .unwrap();
+        CandleTensor::new(match bias {
+            Some(bias) => conv
+                .broadcast_add(&bias.tensor.unsqueeze(1).unwrap())
+                .unwrap(),
+            None => conv,
+        })
+    }
+
     fn conv2d(
         x: FloatTensor<Self, 4>,
         weight: FloatTensor<Self, 4>,
         bias: Option<FloatTensor<Self, 1>>,
         options: ConvOptions<2>,
     ) -> FloatTensor<Self, 4> {
-        todo!()
+        assert!(
+            options.dilation[0] == options.dilation[1]
+                && options.padding[0] == options.padding[1]
+                && options.stride[0] == options.stride[1],
+            "Candle does not support per dimension options in convolutions"
+        );
+        let conv = x
+            .tensor
+            .conv2d(
+                &weight.tensor,
+                options.padding[0],
+                options.stride[0],
+                options.dilation[0],
+                options.groups,
+            )
+            .unwrap();
+        CandleTensor::new(match bias {
+            Some(bias) => conv
+                .broadcast_add(
+                    &bias
+                        .tensor
+                        .unsqueeze(0)
+                        .unwrap()
+                        .unsqueeze(2)
+                        .unwrap()
+                        .unsqueeze(3)
+                        .unwrap(),
+                )
+                .unwrap(),
+            None => conv,
+        })
+    }
+
+    fn conv_transpose1d(
+        x: FloatTensor<Self, 3>,
+        weight: FloatTensor<Self, 3>,
+        bias: Option<FloatTensor<Self, 1>>,
+        options: ConvTransposeOptions<1>,
+    ) -> FloatTensor<Self, 3> {
+        panic!("Candle does not support conv_transpose1d")
     }
```

Comment:
There is a default implementation that uses conv_transpose_2d, so just don't implement this function and it will work!

Reply:
I implemented it on purpose. Because of the asserts I put in conv_transpose2d, we can't call conv_transpose1d from conv_transpose2d as it puts a different second option. So I was thinking about either rewrite conv_transpose1d from conv_transpose2d specifically for candle, or remove the assertions... it was getting confusing, so I chose to wait to see if they are going to write conv_transpose1d (they did write both conv_1d and conv_2d)